deluge/ui/web/json_api.py

```diff
@@ -91,20 +91,18 @@ def __init__(self):
         self._local_methods = {}
         if client.is_classicmode():
             self.get_remote_methods()
-        else:
-            client.disconnect_callback = self._on_client_disconnect
-
-    def on_get_methods(self, methods):
-        """
-        Handles receiving the method names.
-        """
-        self._remote_methods = methods
 
     def get_remote_methods(self, result=None):
         """
         Updates remote methods from the daemon.
+
+        Returns:
+            t.i.d.Deferred: A deferred returning the available remote methods
         """
-        return client.daemon.get_method_list().addCallback(self.on_get_methods)
+        def on_get_methods(methods):
+            self._remote_methods = methods
+            return methods
```

Comment:
Yes, without the return statement a call to the exported connect method will  return None instead os the methods.